spring-cloud-gcp-starters/spring-cloud-gcp-starter-core/src/main/java/org/springframework/cloud/gcp/core/GCPProperties.java

```diff
@@ -27,7 +28,7 @@
 
 	private String projectId;
 
-	private String jsonKeyFile;
+	private Resource credentialKeyFile;
```

Comment:
Actually, I'm thinking it might be confusing for users to have two places to get "credentials" from -- `GcpProperties` and the injected `GoogleCredentials` object (by `GcpContextAutoConfiguration`).

In the following example, the user would get the project ID from `GcpProperties` and the credentials from a different `GoogleCredentials` object, even though `GcpProperties` would contain a credentials field.

```
  @Bean
  @ServiceActivator(inputChannel = "pubsubOutputChannel")
  public MessageHandler messageSender(GCPProperties gcpProperties,
      GoogleCredentials credentials) throws IOException {
    PubSubMessageHandler outboundAdapter =
        new PubSubMessageHandler(gcpProperties.getProjectId(), credentials);
    outboundAdapter.setTopic("test");
    return outboundAdapter;
  }
```

I'm wondering if it's possible to turn this field into `GoogleCredentials`, instead, and perform some kind of magic similar to the `String` -> `Resource` conversion. WDYT @artembilan ?